klippy/extras/load_cell_probe.py

```diff
@@ -0,0 +1,377 @@
+    def _raise_probe(self):
+        toolhead = self.printer.lookup_object('toolhead')
+        start_pos = toolhead.get_position()
+        self.deactivate_gcode.run_gcode_from_command()
+        if toolhead.get_position()[:3] != start_pos[:3]:
+            raise self.printer.command_error(
+                "Toolhead moved during probe deactivate_gcode script")
+
+    def _lower_probe(self):
+        toolhead = self.printer.lookup_object('toolhead')
+        start_pos = toolhead.get_position()
+        self.activate_gcode.run_gcode_from_command()
+        if toolhead.get_position()[:3] != start_pos[:3]:
+            raise self.printer.command_error(
+                "Toolhead moved during probe activate_gcode script")
+
+    def probing_move(self, pos, speed):
+        return self._helper.tapping_move(self, pos, speed)
+
+    def multi_probe_begin(self):
+        self.multi = 'FIRST'
+
+    def multi_probe_end(self):
+        self._raise_probe()
+        self.multi = 'OFF'
+
+    def probe_prepare(self, hmove):
+        if self.multi == 'OFF' or self.multi == 'FIRST':
+            self._lower_probe()
+            if self.multi == 'FIRST':
+                self.multi = 'ON'
+
+    def probe_finish(self, hmove):
+        if self.multi == 'OFF':
+            self._raise_probe()
+
+    def get_position_endstop(self):
+        return self._position_endstop
+
+
+class LoadCellPrinterProbe:
+    def __init__(self, config, load_cell_inst, load_cell_endstop):
+        self.printer = config.get_printer()
+        self.mcu_probe = load_cell_endstop
+        self._load_cell = load_cell_inst
+        self.cmd_helper = probe.ProbeCommandHelper(config, self,
+                                             self.mcu_probe.query_endstop)
+        self.probe_offsets = probe.ProbeOffsetsHelper(config)
+        self.probe_session = probe.ProbeSessionHelper(config, self.mcu_probe)
+
+    # Copy of PrinterProbe methods
+    def get_probe_params(self, gcmd=None):
+        return self.probe_session.get_probe_params(gcmd)
+
+    def get_offsets(self):
+        return self.probe_offsets.get_offsets()
+
+    def start_probe_session(self, gcmd):
+        return self.probe_session.start_probe_session(gcmd)
+
+    def get_status(self, eventtime):
+        status = self.cmd_helper.get_status(eventtime)
+        status.update(self._load_cell.get_status(eventtime))
+        status.update(self.mcu_probe.get_status(eventtime))
+        return status
+
+
+def load_config(config):
+    # Sensor types supported by load_cell_probe
+    sensors = {}
+    sensors.update(hx71x.HX71X_SENSOR_TYPES)
+    sensors.update(ads1220.ADS1220_SENSOR_TYPE)
+    sensor_class = config.getchoice('sensor_type', sensors)
+    sensor = sensor_class(config)
+    lc = load_cell.LoadCell(config, sensor)
+    printer = config.get_printer()
+    name = config.get_name().split()[-1]
+    lc_name = 'load_cell' if name == "load_cell_probe" else 'load_cell ' + name
+    printer.add_object(lc_name, lc)
```

Comment:
I'm not sure what this is trying to do, but it doesn't look right.  In general, every extras module should be registered only once under its full config name (and no other name).  There have been a few cases where we didn't do this, and that has turned into a pain so I'd like to avoid more cases.

Reply:
I till this out. I don't remember exactly what I was worried about here. I think it was something to do with the ambiguity when both a [load_cell_probe] and a [load_cell] are present. You can end up with either one being the "default" but I guess that's OK.